### PR TITLE
TRUNK-5060: globalPropertyChanged catches all exceptions

### PR DESCRIPTION
**Pull request by PKatGITHUB**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5060

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by f4ww4z** [comment]
@PKatGITHUB Did you take a look at https://wiki.openmrs.org/display/docs/Pull+Request+Tips? You'll need to claim the issue first, afterwards request a code review.
Can you also link the JIRA ticket in the PR description?

**Comment by f4ww4z** [comment]
@PKatGITHUB Thank you for your PR :)

You need to click the 'Claim Issue' button on JIRA. Then, click on 'Request a code review'. More info at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by PKatGITHUB** [comment]
@f4ww4z sorry i missed that.
Now i have done it.
Thank you :)

**Comment by dkayiwa** [comment]
@PKatGITHUB did you go through all the JIRA ticket comments?

**Comment by PKatGITHUB** [comment]
@dkayiwa yes. I have gone through the comments and found that there is just a need to remove the try catch block. 
Did i missed something ? 
Thank you.

**Comment by PKatGITHUB** [comment]
@teleivo thank you so much for your help. Yeah i will rebase it surely.
Thank you :)

**Comment by PKatGITHUB** [comment]
@teleivo i rebased it. Does it seem correct?

**Comment by f4ww4z** [comment]
@PKatGITHUB You should end up with 1 commit only.

@teleivo will have a better solution, but considering that you are only removing `try` and `catch` block, you can run `git reset --hard upstream/master`, which will **reset all your changes**. Then, make your change again, commit (`commit -am "PR description"` and finally run `git push -f` to have the latest master changes.

**Comment by teleivo** [comment]
@PKatGITHUB  can you please squash your commits into one, you have 8 now.

Please make sure you have read https://wiki.openmrs.org/display/docs/Pull+Request+Tips as it also contains other helpful tips and things we would like you to do when you open a pull request.

File: api/src/main/java/org/openmrs/api/impl/GlobalLocaleList.java [hunk 1]
```diff
@@ -33,16 +33,10 @@
...
-				}
-			}
-			catch (Exception e) {
-				// bad locale spec? just ignore it. the UI should take care of
-				// guiding the user.
-			}
+			Locale allowedLocale = LocaleUtility.fromSpecification(allowedLocaleString.trim());
+        		if (allowedLocale != null) {
+				allowedLocales.add(allowedLocale);
+                        }
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/api/impl/GlobalLocaleList.java]: @PKatGITHUB can you plese fix the indentation.

https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE

**Comment by PKatGITHUB** [comment]
@teleivo thank you for your kind support. 
@f4ww4z thank you so much for your help.
@teleivo is this ready for merging now? :)

**Comment by PKatGITHUB** [comment]
@teleivo is it fine now?

**Comment by teleivo** [comment]
@PKatGITHUB thank you!

**Comment by PKatGITHUB** [comment]
@teleivo thank you very much for all of your support!
@f4ww4z thank you for your help!
:)

**Pull request merged**